document-store/src/main/java/org/hypertrace/core/documentstore/postgres/utils/PostgresUtils.java

```diff
@@ -223,8 +220,8 @@ public static String parseNonCompositeFilter(
    * part of the below method, we are only using field accessor pattern. This method will be used in
    * Having / Where clause perperation.
    *
-   * <p>See the corresponding test at {@link
-   * PostgresQueryParserTest.testAggregationFilterAlongWithNonAliasFields} In the above example,
+   * <p>See the corresponding test at
+   * PostgresQueryParserTest.testAggregationFilterAlongWithNonAliasFields. In the above example,
```

Comment:
Potential optimisation of methods `prepareParsedNonCompositeFilter` and `parseNonCompositeFilter`. The mapping between the operator name (EQ) and operator (=) can be unified to either a method or a map.

Didn't notice before, but, better to either create the map (in the optimisation mentioned above) directly with `RelationalOperator` as the key. Or, have the switch case on enum values. It makes it easier to catch missing enum mappings (either through a coverage test or a warning in the switch case about missing cases).

Reply:
Yes, I am planning to combine `prepareParsedNonCompositeFilter` + `parseNonCompositeFilter`. As part of it, will take care.